src/Orleans.Runtime/Networking/SiloConnectionFactory.cs

```diff
@@ -47,7 +49,24 @@ protected override Connection CreateConnection(ConnectionContext context)
                 this.messageFactory,
                 this.localSiloDetails,
                 this.siloStatusOracle,
-                this.connectionManager);
+                this.connectionManager,
+                this.ConnectionOptions);
+        }
+
+        private void EnsureInitialized()
+        {
+            if (!isInitialized)
+            {
+                lock (this.initializationLock)
+                {
+                    if (!isInitialized)
+                    {
+                        this.messageCenter = this.serviceProvider.GetRequiredService<MessageCenter>();
```

Comment:
why cant you use  ctor injection for these services?

Reply:
Circular dependency